tests/cancun/eip6780_selfdestruct/test_selfdestruct.py

```diff
@@ -967,112 +865,77 @@ def test_selfdestruct_created_same_block_different_tx(
 
     post: Dict[Address, Account] = {
         entry_code_address: Account(
-            code="0x00",
             storage=entry_code_storage,
         ),
         sendall_recipient_addresses[0]: Account(balance=sendall_amount, storage={0: 1}),
     }
 
     if eip_enabled:
-        post[selfdestruct_contract_address] = Account(
-            balance=0, code=selfdestruct_code, storage={0: call_times}
-        )
+        post[selfdestruct_contract_address] = Account(balance=0, storage={0: call_times})
     else:
         post[selfdestruct_contract_address] = Account.NONEXISTENT  # type: ignore
 
-    nonce = count()
     txs = [
         Transaction(
-            ty=0x0,
-            value=0,
+            value=selfdestruct_contract_initial_balance,
             data=selfdestruct_contract_initcode,
-            chain_id=0x0,
-            nonce=next(nonce),
+            sender=sender,
             to=None,
-            gas_limit=100_000_000,
-            gas_price=10,
-            protected=False,
+            gas_limit=500_000,
         ),
         Transaction(
-            ty=0x0,
-            value=100_000,
+            value=entry_code_balance,
             data=entry_code,
-            chain_id=0x0,
-            nonce=next(nonce),
+            sender=sender,
             to=None,
-            gas_limit=100_000_000,
-            gas_price=10,
-            protected=False,
+            gas_limit=500_000,
         ),
     ]
 
     blockchain_test(genesis_environment=env, pre=pre, post=post, blocks=[Block(txs=txs)])
 
 
-@pytest.mark.parametrize(
-    "selfdestruct_code",
-    [
-        pytest.param(
-            Op.DELEGATECALL(
-                Op.GAS,
-                PRE_EXISTING_SELFDESTRUCT_ADDRESS,
-                0,
-                0,
-                0,
-                0,
-            ),
-            id="delegatecall",
-        ),
-        pytest.param(
-            Op.CALLCODE(
-                Op.GAS,
-                PRE_EXISTING_SELFDESTRUCT_ADDRESS,
-                0,
-                0,
-                0,
-                0,
-                0,
-            ),
-            id="callcode",
-        ),
-    ],
-)  # The self-destruct code is delegatecall
 @pytest.mark.parametrize("call_times", [1])
 @pytest.mark.parametrize("selfdestruct_contract_initial_balance", [0, 1])
+@pytest.mark.parametrize("call_opcode", [Op.DELEGATECALL, Op.CALLCODE])
 @pytest.mark.parametrize("create_opcode", [Op.CREATE])
 @pytest.mark.valid_from("Shanghai")
 def test_delegatecall_from_new_contract_to_pre_existing_contract(
```

Comment:
```suggestion
def test_calling_from_new_contract_to_pre_existing_contract(
```
Pedantic comment but should the `test_delegatecall_from...` be  `test_calling_from...`?